mac-au: Add an AUv3 host as filter

pkviet opened the pull request:
### Description
This adds an AUv3 host to obs-studio.
The host is loaded as a filter.
It is compatible with both v2 and v3 AudioUnits which provide audio filters on macOS.
Sidechain is supported.
Resizing of the AU GUI is also supported (that was tricky).
The AU host follows version 3 specifications, which add a bridge allowing to load AU version 2.

Some screenshots:
**AUv2 from Apple**
<img width="673" height="408" alt="Au1" src="https://github.com/user-attachments/assets/7158b77a-5a23-4da4-bd64-90ba32afcd87" />

**AUv3**
<img width="735" height="391" alt="Au2" src="https://github.com/user-attachments/assets/ca31d943-f3bf-40b3-b8b9-c6d02285a453" />

**Sidechain**
<img width="437" height="301" alt="Au3" src="https://github.com/user-attachments/assets/3fe42696-ce68-48f5-8996-ef54ac52e52d" />

**Sidechain and GUI**
<img width="676" height="383" alt="Au4" src="https://github.com/user-attachments/assets/25ff8d2c-f7a0-44b8-9a38-d259860005f7" />

### Motivation and Context
AU filters are native to macOS. They're a very good alternative to VST on macOS.
macOS ships with a series of good AU filters.

I coded a VST3 host and decided to look at the competition (AU & CLAP).
It seemed easy to write an AU host so for fun, I did it. The architecture is the same as that of obs-vst3.

### How Has This Been Tested?
Tested on macOS 26.1.
AU both v2 & v3 work fine, with or without GUI.
Tested the GUIs resize fine.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

akagumo commented:
sometimes macOS doesn't record the crash unfortunately. so no logs in the console.

I took a break and came back and IK plugins are not crashing now.
the crashing of the IK AU plugins might have been due to having two vst2 plugins on that were not installed.
I might have installed the newer version of that plugin which is only vst3 now, so no vst2 version.
I think I turned them off before break. testing more plugins now.
will be great when vst3 and AU are finalized into official release of OBS!

pkviet committed:
mac-au: Add an AUv3 host as filter

This adds an AUv3 host to obs-studio.
The host is loaded as a filter.
It is compatible with both v2 and v3 AudioUnits.
Sidechain is supported.

Signed-off-by: pkv <pkv@obsproject.com>

pkviet commented:
Closing since VST3 is enough. Don't want to have to maintain too much stuff.
